packages/@angular/cli/update/check.ts

```diff
@@ -0,0 +1,143 @@
+  const threshold = convertFrequency(freq);
+  return diff > threshold;
+}
+
+export function newVersionAvailable(
+  rootDir: string,
+  name: string,
+  options?: CheckOptions): Promise<CheckResult> {
+  return Promise.all([
+      getLocalVersion(rootDir, name),
+      getRemoteVersion(name, options)
+    ])
+    .then(([localVersion, remoteVersion]) => {
+      const newer = lt(localVersion, remoteVersion);
+      const result: CheckResult = {
+        currentVersion: localVersion
+      };
+      if (newer) {
+        result.newVersion = remoteVersion;
+      }
+      return result;
+    });
+}
+
+function encodePackageName(name: string): string {
+  let encoded = encodeURIComponent(name);
+  if (name.startsWith('@')) {
+    encoded = `@${encoded.substr(3)}`;
+  }
+  return encoded;
+}
+
+function getRemoteVersion(name: string, options?: CheckOptions): Promise<string> {
+  const url = `https://registry.npmjs.org/${encodePackageName(name)}`;
+  return httpsGet(url)
+    .then(resp => {
+      const version = options.includeUnstable ?
+        resp['dist-tags'].next :
+        resp['dist-tags'].latest;
+      return version;
+    });
+}
+
+function getLocalVersion(rootDir: string, name: string): Promise<string> {
+  const packagePath = path.resolve(rootDir, 'node_modules', name, 'package.json');
+  return new Promise((resolve, reject) => {
+    try {
+      const packageInfo = require(packagePath);
+      resolve(packageInfo.version);
+    } catch (err) {
+      reject();
+    }
+
+  });
+}
+
+function httpsGet(url: string): Promise<any> {
+  return new Promise((resolve, reject) => {
+    https.get(url, (res) => {
+      const { statusCode } = res;
+      const contentType = res.headers['content-type'];
+
+      let error;
+      if (statusCode !== 200) {
+        error = new Error(`Request Failed. Status Code: ${statusCode}`);
+      } else if (!/^application\/json/.test(contentType)) {
+        error = new Error(
+          `Invalid content-type. Expected application/json but received ${contentType}`);
+      }
+      if (error) {
+        reject(error.message);
+        // consume response data to free up memory
+        res.resume();
+        return;
+      }
+
+      res.setEncoding('utf8');
+      let rawData = '';
+      res.on('data', (chunk) => {
+        rawData += chunk;
+      });
+
+      res.on('end', () => {
+        try {
+          const parsedData = JSON.parse(rawData);
+          resolve(parsedData);
+        } catch (e) {
+          reject(e.message);
+        }
+      });
+    }).on('error', (e) => {
+      reject(e.message);
+    });
+  });
+}
+
+function convertFrequency(frequency: string): number {
+  const value = parseFloat(frequency.substring(0, frequency.length - 1));
+  const unit = frequency[frequency.length - 1];
+  let factor = 1;
+  switch (unit.toLowerCase()) {
+    case 'h':
+      factor = ONE_HOUR;
+      break;
+    case 'w':
+      factor = ONE_WEEK;
+      break;
+    default:
```

Comment:
This is easily achievable by using `30d`/`60d` without having to take into account the number days in a month which is an unnecessary complication for this feature IMHO.